packages/common/infra/src/sync/indexer/README.md

```diff
@@ -0,0 +1,147 @@
+# index
+
+Search engine abstraction layer for AFFiNE.
+
+## Using
+
+1. Define schema
+
+First, we need to define the shape of the data. Currently, there are the following data types.
+
+- 'Integer'
+- 'Boolean'
+- 'FullText': for full-text search, it will be tokenized and stemmed.
+- 'String': for exact match search, e.g. tags, ids.
```

Comment:
Maybe better to define a type for available type strings, and provide a method to define a schema(such as `defineSchema()`) to get better type hints and type safety?

Reply:
Maybe like:

```ts
type IndexType = 'Boolean' | 'String' | 'FullText' | 'Integer';
type IndexSchema<Fields extends string> = Record<Fields, IndexType>;

const defineSchema = <F extends string>(schema: IndexSchema<F>) =>
  schema;

const schema = defineSchema({
  title: 'FullText',
  tag: 'String',
  size: 'Integer',
});
```